training/platform-services/PLATFORM.md

```diff
@@ -0,0 +1,182 @@
+
+> Vault secures, stores and tightly controls access to tokens, passwords, certificates, encryption keys for protecting secrets and other sensitive data
+
+Vault manage secrets and stores the secrets in [dp-configs][dp-configs-secrets] in a secure way and made available to the service (which can turn feature on and off). In general, Vault is used for the following two functionalities:
+1. Storing application configs (secrets) from [dp-configs][dp-configs-secrets]
+2. Storing per-file encryption keys for CMD uploads, image uploads, etc. 
+   
+Some apps that deal with S3 uploads (florence, dataset exporter etc) need a vault client in the app and hence this is informed in their Makefile) while other apps only retrieve configs. Please note that all apps need their [vault policies][vault-policy] defined in [dp-setup/ansible/files/vault-policies][dp-setup-vault-policies]. Policies provide a declarative way to grant or forbid access to certain paths and operations in Vault. That’s why if you look at the vault policies, most will only have one path while some (such as [florence vault policy][florence-vault-policy] which deals with S3 uploads) will have multiple.
+
+More information about Vault can be found on their website - https://www.vaultproject.io/
+
+## Grafana
+Visit the following to access Grafana
+- DEVELOP:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;https://grafana.develop.onsdigital.co.uk
+- PRODUCTION:&nbsp;&nbsp;&nbsp;https://grafana.production.onsdigital.co.uk
+
+[<img src="images/grafana-dev-home.png" title="Grafana Develop Home" width=49.7% height=49.7%/>][grafana-dev-home] [<img src="images/grafana-dev-nomad-task.png" title="Grafana Develop Nomad Task" width=49.7% height=49.7%/>][grafana-dev-nomad-tasks]
+
+Grafana informs how a process is behaving on the machine (its use of CPU, memory, use of IO operations, etc). In Grafana, the [Nomad Task][grafana-dev-nomad-tasks] dashboard is commonly used, which gives a list of tasks (each related to a specific application). From there, you can see the behaviours of specific applications to the machine such as memory usage (e.g. to identify a potential memory leak by changing the range of time to 7 days and analysing the graph). Grafana assists in debugging and we can look at [Kibana](#kibana) to see logs at a particular time based on any anomalies discovered from graphs in Grafana. Furthermore, the [Node Stats 2][grafana-dev-node-stats-2] dashboard is used as well which is specific to EC2 instances performance.
+
+Please note that CPU can spike as reserved CPU can exceed the limit because of its soft limit (CPU bursting). On the other hand, memory usage cannot spike as a hard limit is set for reserved memory and if memory exceeds its limit, the service will stop and restart
```

Comment:
Technically, they can both spike, in that a spike it a rapid change in use. The issue is whether that use exceeds limits. So maybe this could be better phrased around 'CPU can exceed its defined limits (known as bursting), as they are 'soft limits' Memory has a 'hard limit' and any use in excess will cause the service to stop/restart' or something?

Reply:
Agreed and done